testing_and_setup/compass/manage_regression_suite.py

```diff
@@ -419,29 +640,34 @@ def get_test_case_procs(suite_tag):  # {{{
                                     threads = int(threads_str)
                                 except (KeyError, ValueError):
                                     threads = 1
-
                     del config_root
                     del config_tree
-            testcases[test_name] = {'core': test_core,
-                                    'configuration': test_configuration,
-                                    'resolution': test_resolution,
-                                    'test': test_test,
-                                    'path': test_path,
-                                    'procs': procs,
-                                    'threads': threads,
-                                    'prereqs': prereqs}
-
-    return testcases  # }}}
-
-
-def summarize_suite(testcases):  # {{{
+            data[test_name] = {'core': test_core,
+                               'configuration': test_configuration,
+                               'resolution': test_resolution,
+                               'test': test_test,
+                               'path': test_path,
+                               'procs': procs,
+                               'threads': threads,
+                               'prereqs': prereqs}
+            if not data[test_name]['prereqs'] == []:
+                for sub in data[test_name]['prereqs']:
+                    name = sub['name']
+                    if name not in data_prereq:
+                        data_prereq.append(name)
+    return data
```

Comment:
Since `data` gets modified inplace, it's not necessary to return it but it doesn't do any harm.  Previously, I was creating the dictionary `testcases` in this function so it needed to be returned.